src/Budget.php

```diff
@@ -887,13 +887,11 @@ public function showValuesByEntity()
         echo "<tr class='tab_bg_1 noHover'>";
         echo "<td class='right' colspan='" . ($colspan - 1) . "'>" . __('Total spent on the budget') . "</td>";
         echo "<td class='numeric b'>" . Html::formatNumber($total) . "</td></tr>";
-        if ($_SESSION['glpiactive_entity'] == $budget->fields['entities_id']) {
```

Comment:
Imagine the following situation:
 - you have two entities that corresponds to the following tree: `A` > `A.1`;
 - the budget is declared in `A` and visible in child entities;
 - some items that are part of the budget are declared in entity `A` but are NOT visible in child entities;
 - some items that are part of the budget are declared in entity `A.1`.

If your active entity is `A`, and if you have access to all sub-entities, the remaining budget will be correct, as `$total` will correspond to the sum of all items of the entity `A` and all its sub-entities that are related to budget.

If your active entity is `A.1`, the remaining budget will not be correct, as `$total` will ONLY correspond to the sum of all items of the entity `A.1` that are related to budget (items from the entity `A` will not be part of the `$total`).

Note that with or without your patch, if the active entity is `A` but sub entities are not displayed, items of the sub-entity `A.1` will not be part of the `$total`, so the displayed value will be wrong.

I do not know what it the best approach here. We should probably not display names of entities that are not supposed to be visible according to `$_SESSION['glpiactiveentities']`. Maybe the solution is to add a `another entities` entry in the table that will contains the sum of items that are declared in entities that are not excluded by `getEntitiesRestrictCriteria()`. I do not know if it is acceptable to display the sum of "invisible items".

@orthagh, what dio you think?

Reply:
> Maybe the solution is to add a another entities entry in the table 

Agree with that

Reply:
With the last commit, is it OK for you?

Reply:
I think we should add a new line on the table with:
 - `Other entities` in fist column;
 - no value in columns corresponding to details by itemtype,
 - `$total - $subtotal` in the total column.

It would make things a little more clear. Indeed, I am not sure than `active entities` VS `all entities` will be clear for everyone.